### PR TITLE
refactor: generate type definitions so we can use it in other project

### PR DESCRIPTION
**Pull request by BANG88**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,5 +25,8 @@
   "devDependencies": {
     "rollup": "^0.52.1",
     "tape": "^4.3.0"
+  },
+  "dependencies": {
+    "typescript": "^2.7.1"
```
Comment (afc163) [inline package.json]: typescript should be put in `devDependencies`.

Comment (BANG88) [reply package.json]: miss

File: package.json [hunk 2]
```diff
@@ -28,6 +24,7 @@
   "homepage": "https://github.com/afc163/array-tree-filter#readme",
   "devDependencies": {
     "rollup": "^0.52.1",
-    "tape": "^4.3.0"
+		"tape": "^4.3.0",
+		"typescript": "^2.7.1"
```
Comment (afc163) [inline package.json]: wrong indent

**Comment by afc163** [comment]
2.1.0

**Pull request merged**